File: JavaScript/13-罗马数字转整数.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
    let maps = {"I":1,"V":5,"X":10,"L":50,"C":100,"D":500,"M":1000};
    let res = 0
    for (let i = 0;i < s.length;i++ ){
        if ((i < s.length -1)&&(maps[s[i]] < maps[s[i + 1]])){
           res -= maps[s[i]]
        } else {
          res += maps[s[i]] 
        }
    }
    return res
};

File: Go/13-罗马数字转整数.go
package main
func romanToInt(s string) int {
	maps := make(map[string]int)
	maps["I"] = 1
	maps["V"] = 5
	maps["X"] = 10
	maps["L"] = 50
	maps["C"] = 100
	maps["D"] = 500
	maps["M"] = 1000
	res := int(0)
	for i := 0; i< len(s);i++ {
		if i < len(s)-1 && maps[string(s[i])] < maps[string(s[i+1])] {
			res -= maps[string(s[i])]
		} else {
			res += maps[string(s[i])]
		}
	}
	return res
}

